pkg/pluginx/command.go

```diff
@@ -0,0 +1,66 @@
+package pluginx
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/alexflint/go-arg"
+	"github.com/mattn/go-shellwords"
+)
+
+// ParseCommand processes a given command string and stores the result in a given destination.
+// Destination MUST be a pointer to a struct.
+func ParseCommand(pluginName, command string, destination any) error {
+	p, err := arg.NewParser(arg.Config{}, destination)
+	if err != nil {
+		return fmt.Errorf("while creating parser: %w", err)
+	}
+
+	rawCmd := strings.TrimSpace(command)
+	rawCmd = strings.TrimPrefix(rawCmd, pluginName)
+	err = p.Parse(strings.Fields(rawCmd))
+	if err != nil {
+		return fmt.Errorf("while parsing input command: %w", err)
+	}
+
+	return nil
+}
+
+// ExecuteCommand is a simple wrapper around exec.CommandContext to simplify running a given
+// command.
+func ExecuteCommand(ctx context.Context, rawCmd string) (string, error) {
+	var stdout, stderr bytes.Buffer
+
+	envs, args, err := shellwords.ParseWithEnvs(rawCmd)
+	if err != nil {
+		return "", err
+	}
+
+	if len(args) < 1 {
+		return "", fmt.Errorf("invalid raw command: %q", rawCmd)
+	}
+
+	if err := os.Setenv("PATH", fmt.Sprintf(`%s:%s`, pluginsDir, os.Getenv("PATH"))); err != nil {
```

Comment:
I don't like modifying `PATH` - remember that this will be modified also for other executors  like the built-in `kubectl` one. Also, the change is long-living, and for each occurrence you add the same path over and over again.